src/index.js

```diff
@@ -4,143 +4,116 @@ process.env.SENTRY_DSN =
-      }
-      return rq(loginOptions)
-    })
-    .then($ => {
-      log('info', 'Successfully logged in.')
-      const badLogin = $('.error-icon').length > 0
-      if (badLogin) {
-        throw new Error('LOGIN_FAILED')
-      }
-
-      const $txtindispo = $('.txt-indispo')
-      if ($txtindispo.length) {
-        log(
-          'error',
-          $txtindispo
-            .text()
-            .trim()
-            .replace(/\n/g, ' ')
-            .replace(/ */, ' ')
+async function logIn({ login, password }) {
+  await signin({
+    url: 'https://www.mon-compte.bouyguestelecom.fr/cas/login',
+    formSelector: 'form',
+    formData: { username: login, password },
+    validate: (statusCode, $) => {
+      if (
+        $.html().includes(
+          'Votre identifiant ou votre mot de passe est incorrect'
         )
-        throw new Error('VENDOR_DOWN')
+      ) {
+        return false
+      } else {
+        return true
       }
-    })
-    .then(() => {
-      log('info', 'Download bill HTML page...')
-      return rq(billUrl)
-    })
-}
-
-// Procedure to extract bill data from the page.
-function parsePage($) {
-  let baseDlUrl =
-    'https://www.bouyguestelecom.fr/parcours/facture/download/index'
-  const entries = []
-  moment.locale('fr')
-
-  // We browse the bills table by processing each line one by one.
-  $('.download-facture').each(function() {
-    // Markup is not clean, we grab the date from the tag text.
-    let date = $(this)
-      .text()
-      .trim()
-      .split(' ')
-      .splice(0, 2)
-      .join(' ')
-      .trim()
-
-    // Amount is in a dirty field. We work on the tag text to extract data.
-    let amount = $(this)
-      .find('.small-prix')
-      .text()
-      .trim()
-    amount = parseFloat(amount.replace('€', ',').replace(',', '.'))
-
-    // Get the facture id and build the download url from it.
-    const id = $(this).attr('facture-id')
-    const params = { id }
-    const fileurl = `${baseDlUrl}?${qs.stringify(params)}`
-    date = moment(date, 'MMMM YYYY').add(14, 'days')
-
-    // Build bill object.
-    const bill = {
-      vendor: 'Bouygues Box',
-      date: date.toDate(),
-      amount,
-      fileurl,
-      filename: getFileName(date)
     }
-    entries.push(bill)
   })
+  // Acredite token for downloading file via the API
+  const resp = await rq(
+    'https://oauth2.bouyguestelecom.fr/authorize?client_id=a360.bouyguestelecom.fr&response_type=id_token%20token&redirect_uri=https%3A%2F%2Fwww.bouyguestelecom.fr%2Fmon-compte%2F',
+    {
+      resolveWithFullResponse: true
+    }
+  )
 
-  log('info', 'Bill data parsed.')
+  const href = resp.request.uri.href.split('=')
+  const accessToken = href[1].split('&')[0]
+  rq = rq.defaults({
+    headers: {
+      Authorization: `Bearer ${accessToken}`
+    }
+  })
+  const idPersonne = jwt(href.pop()).id_personne
+  return idPersonne
+}
 
-  return Promise.resolve(entries)
+function foundLigneType(idCompte, contrats) {
```

Comment:
findLineType ?

Reply:
Most of variable name in the API are french worded. So i think it's better to keep this like this, as it more representative of fields in the api jsons.
'contratsSignes', 'comptes', 'ligneType'

Reply:
dans ce cas findLigneType ou trouverLigneType ?

Reply:
C'est pas très important bien sûr

Reply:
Ah oui va, pour findLigneType, j'avais pas vu le participe passé anglais.